### ISSUE TITLE
Rewrite `synthetic.py` such that all index variables run from 0, 1, 2, 3, ...

### ISSUE THREAD

**Issue by ChristianZimpelmann** [iza-institute-of-labor-economics/gettsim#365]
### Current and desired situation

- The implementation currently loops over `dimensions` of the heterogeneous variables. 
- `tu_id` and `hh_id` are created such that for more than one dimension some indices start at 100.

```python
    df["hh_id"] = 100 * dim + df.index
    df["tu_id"] = 100 * dim + df.index
```

This leads to two problems:
- There are gaps in the indices
- If a dimension leads to more than 100 households, some individuals are grouped in the same households that are not expected to be.

**Comment by hmgaudecker** [comment]
I don't think that having no gaps is required. Actually a good case to check that Jax will not get things wrong eventually :smile:

Previous situation just made it look like the 100 actually means something.

So I would suggest to just close this one.

**Comment by ChristianZimpelmann** [comment]
I adjusted the description as it probably wasn't clear enough. I think we need to address this eventually so I would like to keep it open.

**Comment by ChristianZimpelmann** [comment]
Closed by #362
